test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+                .short("r")
+                .long("enable-value-recorder")
+                .help("Enables or disables value recorder instrument.")
+        )
+        .arg(
+            Arg::with_name("enable-value-observer")
+                .short("v")
+                .long("enable-value-observer")
+                .help("Enables or disables value observer instrument.")
+        )
+        .get_matches();
+
+    let config = Config {
+        update_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_UPDATE_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("update-rate"), f64).unwrap_or(1.0)
+        },
+        push_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_PUSH_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("push-rate"), f64).unwrap_or(0.2)
+        },
+        batching_enabled: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_BATCHING_ENABLED") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("batching-enabled"), bool).unwrap_or(false)
+        },
+        otlp_endpoint: if let Ok(v) = std::env::var("OBS_AGENT_OTLP_ENDPOINT") {
+            v.parse()?
+        } else {
+            matches
+                .value_of("otlp-endpoint")
+                .unwrap_or("http://localhost:4317")
+                .to_string()
+        },
+        enable_value_recorder: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_RECORDER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-recorder"), bool).unwrap_or(false)
+        },
+        enable_value_observer: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_OBSERVER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-observer"), bool).unwrap_or(false)
+        },
+    };
+    println!("{:?}", config);
+    Ok(config)
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let config = init_config()?;
+    let _started = init_meter(1.0 / config.push_rate, config.otlp_endpoint)?;
+    let meter = global::meter("microsoft.com/azureiot-edge");
+
+    // Init synchronous instruments
+    let counter = meter.u64_counter("u64_counter_example").init();
+    let ud_counter = meter
+        .i64_up_down_counter("i64_up_down_counter_example")
+        .init();
+    let mut value_recorder: Option<ValueRecorder<f64>> = None;
+    if config.enable_value_recorder {
+        value_recorder = Some(
+            meter
+                .f64_value_recorder("f64_value_recorder_example")
+                .init(),
+        );
+    }
+
+    // Init asynchronous instruments
+    let sum = Arc::new(Mutex::new(0));
+    let sum_clone = Arc::clone(&sum);
+    let sum_observer_cb = move |res: ObserverResult<u64>| {
+        let sum_clone = sum_clone.lock().unwrap();
```

Comment:
Can this unwrap fail?

Reply:
`lock()` can fail if the lock is poisoned (meaning a thread panicked previously while holding the lock), It also can panic if the lock is already held by the current thread.

I decided to replace `lock()` with `try_lock()` which doesn't block or panic.  I also got rid of the unwrap().